cmd/zc_traverser_local.go

```diff
@@ -105,12 +114,18 @@ func UnfurlSymlinks(symlinkPath string) (result string, err error) {
 
 			result = common.ToExtendedPath(result)
 
+			// Either we can store all the symlink seen till now for this path or we count how many iterations to find out cyclic loop.
+			if _, ok := seenSymlinkPath[result]; ok {
+				return "", errors.New("failed to unfurl symlink: too many links")
+			}
```

Comment:
We can simply count the number of loops encountered (as you also mention).
That's simpler and more robust. Linux also does the simple loop test and fails any symlink loops larger than 40.
We can also pick similar number. Anyways links more than that deep would be absurd.